src/pint/models/stand_alone_psr_binaries/DDH_model.py

```diff
@@ -0,0 +1,165 @@
+    decomposition used in :class:`pint.models.stand_alone_psr_binaries.ELL1H_model.ELL1Hmodel`.
+
+    References
+    ----------
+    - Freire and Wex (2010), MNRAS, 409, 199 [1]_
+
+    .. [1] https://ui.adsabs.harvard.edu/abs/2010MNRAS.409..199F/abstract
+
+
+
+    """
+
+    def __init__(self, t=None, input_params=None):
+        super().__init__()
+        self.binary_name = "DDH"
+        self.param_default_value.update(
+            {
+                "H3": 0.0 * u.second,
+                "STIGMA": 0.0 * u.Unit(""),
+            }
+        )
+        self.binary_params = list(self.param_default_value.keys())
+        self.set_param_values()  # Set parameters to default values.
+
+    @property
+    def SINI(self):
+        return 2 * self.STIGMA / (1 + self.STIGMA**2)
+
+    @property
+    def M2(self):
+        return self.H3 / self.STIGMA**3 / Tsun.value
+
+    @SINI.setter
+    def SINI(self, val):
+        log.debug(
+            "DDH model uses H3/STIGMA as Shapiro delay parameter. SINI will not be used."
+        )
+
+    @M2.setter
+    def M2(self, val):
+        log.debug(
+            "DDH model uses H3/STIGMA as Shapiro delay parameter. M2 will not be used."
+        )
+
+    def d_delayS_d_par(self, par):
+        """Derivative.
+
+        Computes::
+
+            dsDelay/dPar = dsDelay/dH3*dH3/dPar+
+                        dsDelay/decc*decc/dPar+
+                        dsDelay/dE*dE/dPar+
+                        dsDelay/domega*domega/dPar+
+                        dsDelay/dSTIGMA*dSTIGMA/dPar
+        """
+        e = self.ecc()
+        cE = np.cos(self.E())
+        sE = np.sin(self.E())
+        sOmega = np.sin(self.omega())
+        cOmega = np.cos(self.omega())
+        TM2 = self.M2.value * Tsun
+
+        logNum = (
+            1
+            - e * cE
+            - self.SINI * (sOmega * (cE - e) + (1 - e**2) ** 0.5 * cOmega * sE)
+        )
+        with u.set_enabled_equivalencies(u.dimensionless_angles()):
+            dH3_dpar = self.prtl_der("H3", par)
+            dsDelay_dH3 = -2 * np.log(logNum) / self.STIGMA**3
+            dSTIGMA_dpar = self.prtl_der("STIGMA", par)
+            dsDelay_dSTIGMA = 6 * self.H3 * np.log(
+                logNum
+            ) / self.STIGMA**4 - 2 * self.H3 * (
+                -2
+                * (sOmega * (cE - e) + (1 - e**2) ** 0.5 * cOmega * sE)
+                / (1 + self.STIGMA**2)
+                + 4
+                * self.STIGMA**2
+                * (sOmega * (cE - e) + (1 - e**2) ** 0.5 * cOmega * sE)
+                / (1 + self.STIGMA**2) ** 2
+            ) / (
+                self.STIGMA**3
+                * (
+                    1
+                    - e * cE
+                    - (
+                        2
+                        * self.STIGMA
+                        * (sOmega * (cE - e) + (1 - e**2) ** 0.5 * cOmega * sE)
+                    )
+                    / (1 + self.STIGMA**2)
+                )
+            )
```

Comment:
✓

Reply:
Did you use Mathematica/Maple for this? Can you paste a screenshot of the equation as a comment for posterity?

Reply:
Posted below